contracts/base/BlockTimestamp.sol

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+/// @title Function for getting block timestamp
+/// @dev Base contract that is overridden for tests
+abstract contract BlockTimestamp {
+    /// @dev Method that exists purely to be overridden for tests
+    /// @return The current block timestamp
+    function _blockTimestamp() internal view virtual returns (uint256) {
```

Comment:
I think now that we work in forge this shouldnt be needed?
you can do `vm.warp(timestamp)` in a test, and forge will set the current timestamp to that :)